src/ralph/back_office/models.py

```diff
@@ -297,3 +299,13 @@ def return_report(self, request, **kwargs):
         self._generate_report(name='return', request=request)
 
     return_report.verbose_name = _('Return report')
+
+
+@receiver(pre_save, sender=BackOfficeAsset)
+def hostname_assigning(sender, instance, raw, using, **kwargs):
+    """A hook for assigning ``hostname`` value when an
+    back office asset is edited.
+    """
+    if getattr(settings, 'BACK_OFFICE_ASSET_AUTO_ASSIGN_HOSTNAME', None):
+        if instance.status == BackOfficeAssetStatus.in_progress:
```

Comment:
what if asset already has `in_progress` status? we don't want to change hostname in this case. You should fetch previous asset from db and check if status field has changed